src/network/serverpackethandler.cpp

```diff
@@ -380,55 +380,47 @@ void Server::handleCommand_ClientReady(NetworkPacket* pkt)
 {
 	session_t peer_id = pkt->getPeerId();
 
-	PlayerSAO* playersao = StageTwoClientInit(peer_id);
-
-	if (playersao == NULL) {
-		errorstream << "TOSERVER_CLIENT_READY stage 2 client init failed "
-			"peer_id=" << peer_id << std::endl;
-		DisconnectPeer(peer_id);
-		return;
-	}
-
-
-	if (pkt->getSize() < 8) {
-		errorstream << "TOSERVER_CLIENT_READY client sent inconsistent data, "
-			"disconnecting peer_id: " << peer_id << std::endl;
-		DisconnectPeer(peer_id);
-		return;
-	}
-
+	// decode all information first
 	u8 major_ver, minor_ver, patch_ver, reserved;
+	u16 formspec_ver;
```

Comment:
Needs initialization to 1 if the packet lacks this information (Minetest <5.1.0).